### ISSUE TITLE
SMW\SQLStore\EntityStore\IdChanger->change() creates duplicate smw_di_wikipage rows

### ISSUE THREAD

**Issue by hexmode** [SemanticMediaWiki/SemanticMediaWiki#5083]
### Issue

While running rebuildData.php RedirectUpdater's [moveAsRedirect()](https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/91546f4c02af0409f97bacecd142a6170d78cde2/src/SQLStore/RedirectUpdater.php#L561) calls [IdChanger->change()](https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/91546f4c02af0409f97bacecd142a6170d78cde2/src/SQLStore/EntityStore/IdChanger.php#L166) to update the ids in `swm_di_wikipage`.

This causes problems on when you [add a hook to create indexes](https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/docs/examples/hook.sqlstore.installer.beforecreatetablescomplete.md) which is necessary for [some databases](https://www.percona.com/doc/percona-xtradb-cluster/LATEST/features/pxc-strict-mode.html#tables-without-primary-keys) since the index suggested for `smw_di_wikipage` is across all columns of the table and the update described above can result in duplicate rows.

**Comment by hexmode** [comment]
The problem is caused by using the suggested index entry for the wikipage table (`'smw_di_wikipage' => 'p_id,s_id,o_id',`).

This seemed to make sense since those are the only columns in that table.  If it has conflicting keys, then there must be duplicate rows, right?

So, with that in mind, I attempted to change `idChanger->change()` so that it simply deleted the row rather than changing the id if an identical id already existed.

That meant, though,  that some properties were not propagated correctly.  Pages that had data showing up on them were suddenly missing data.  One new subobject I created seemed to disappear completely.

With this in mind, I updated the `smw_di_wikipage` table so that it had an otherwise unused auto incrementing `id` column to provide the needed primary key.  This appears to work (so far).
